Clean up lowering APIs

mcremon-meta opened the pull request:
Summary: Changes some types in `export_to_edge`, add an `export_to_cadence` function, and refactor some calls. Mostly preemptive work before adding a lot of AoT passes.

Differential Revision: D57579738

dulinriley commented on backends/cadence/aot/compiler.py:
```diff
@@ -51,12 +58,32 @@ def export_to_edge(
-        expo_program, compile_config=EdgeCompileConfig(_check_ir_validity=False)
+        expo_program,
+        compile_config=EdgeCompileConfig(
+            _check_ir_validity=False, _skip_dim_order=True
```
What's this added `_skip_dim_order` kwarg for?

dulinriley commented on backends/cadence/aot/compiler.py:
```diff
@@ -51,12 +58,32 @@ def export_to_edge(
+    inputs: Tuple[Any, ...],
+    dump_graphs: bool = False,
+) -> EdgeProgramManager:
+    edge_program_manager = export_to_edge(model, inputs)
```
Should this just take in an EdgeProgramManager that separately had `export_to_edge` called so that the APIs can be combined?

dulinriley commented on backends/cadence/aot/utils.py:
```diff
@@ -71,29 +71,15 @@ def get_ops_count(graph_module: torch.fx.GraphModule) -> Dict[str, int]:
     to_edge_ops_count = get_ops_count(to_edge_gm)
     jarvis_ops_count = get_ops_count(jarvis_gm)
 
-    # De-duplicate the "<op>" and "<op>_copy" ops
```
Why remove this part?

mcremon-meta committed:
Clean up lowering APIs (#3690)

Summary:

Changes some types in `export_to_edge`, add an `export_to_cadence` function, and refactor some calls. Mostly preemptive work before adding a lot of AoT passes.

Reviewed By: dulinriley, zonglinpengmeta

Differential Revision: D57579738
